server/src/intercepter/response.intercepter.ts

```diff
@@ -0,0 +1,55 @@
+import {
+  Injectable,
+  NestInterceptor,
+  ExecutionContext,
+  BadGatewayException,
+  CallHandler,
+  HttpException,
+  HttpStatus,
+} from '@nestjs/common';
+import { map, catchError } from 'rxjs/operators';
+import { Observable, throwError } from 'rxjs';
+
+// Response by routes 
+@Injectable()
+export class ResponseInterceptor<T> implements NestInterceptor<T, { data: T }> {
+  intercept(
+    context: ExecutionContext,
+    next: CallHandler,
+  ): Observable<{ data: T }> {
+    return next.handle().pipe(map((data) => ({ data })));
```

Comment:
is there any impact in response after using response Interceptor and without interceptor?

Reply:
yes while using this our all response data is getting in 

```
{
  data : {
    // response 
  }

}
```

and in error one it responde error : { }